azure-iot-device/setup.py

```diff
@@ -27,13 +27,18 @@
 with open("README.md", "r") as fh:
     _long_description = fh.read()
 
-constant = {}
-with open("azure/iot/device/constant.py") as fh:
-    exec(fh.read(), constant)
+
+version = None
+with open("azure/iot/device/constant.py", "r") as fh:
+    for line in fh:
+        if "VERSION" in line:
+            constant, value = line.strip().split("=")
```

Comment:
This could turn up a false positive if the line contains "VERSION" i.e. we introduce some new variable that also contains the word, which would be bad.

I wonder if it might be better to consider having a separate `version.txt` file that maintains the version that we can read safely here (and not have to deal with any special logic to exclude other values), and then also read the same file into a variable `constant.py` for use elsewhere in the codebase.

Something similar to example 4 in the Single Source Packaging guidelines.
[https://packaging.python.org/guides/single-sourcing-package-version/](https://packaging.python.org/guides/single-sourcing-package-version/)

I don't love this solution either though because I hate the idea of having an extra file.

Reply:
Oliva did mention this earlier, but I didn't like have versions in two places.  There needs to be one source of truth for versions.

Reply:
It would actually still be a single source of truth. Version would only be maintained in `version.txt`, and would be dynamically imported from that single source into `constant.py` (where the non-version constants are defined), and then all code references would refer to `constant.py`

---
In reply to: [389742007](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r389742007) [](ancestors = 389742007)

Reply:
there should not be naming conflict right now as i shifted to regex...precisely for this reason...

---
In reply to: [391159201](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r391159201) [](ancestors = 391159201,389742007)